language/vm/vm_runtime/vm_runtime_types/src/loaded_data/struct_def.rs

```diff
@@ -52,11 +65,17 @@ impl CanonicalDeserialize for StructDef {
     where
         Self: Sized,
     {
-        // Libra only runs on 64-bit machines.
-        let num_defs = deserializer.decode_u64()? as usize;
-        let field_defs: Result<_> = (0..num_defs)
-            .map(|_| Type::deserialize(deserializer))
-            .collect();
-        Ok(StructDef::new(field_defs?))
+        match deserializer.decode_u8()? {
+            0x0 => {
+                // Libra only runs on 64-bit machines.
+                let num_defs = deserializer.decode_u64()? as usize;
+                let field_defs: Result<_> = (0..num_defs)
+                    .map(|_| Type::deserialize(deserializer))
+                    .collect();
+                Ok(StructDef::new(field_defs?))
+            }
+            0x1 => Ok(StructDef::Native(deserializer.decode_struct()?)),
```

Comment:
Would it make more sense to have a tag for each different kind of native struct? Using `0x1` for native + a sub-tag (i.e., `NativeStructTag) of `0` for `Vector` seems to add a fair amount of indirection (especially given that we aren't likely to have many native structs).

Reply:
That makes sense. However IMHO in won't really matter that much tho as we are not using this serialization for any on-chain operations. This serialization was used by fuzzer only.